src/odemis/acq/stream/_projection.py

```diff
@@ -218,6 +223,437 @@ def _updateImage(self):
+        logging.debug("No background image with polarization mode %s ." % pol_mode)
+        return None
+
+    def _processBackground(self, data, pol_mode):
+        """
+        Process the background on the raw data. Try to get the background image on the stream
+        and substracts it if available. Otherwise process do a simple background processing.
+        :param data: (DataArray) The data that will be background corrected.
+        :param pol_mode: (str) The polarization mode, the background is requested for.
+        :return: (DataArray) The background corrected data.
+        """
+        bg_image = self._getBackground(pol_mode)
+        if bg_image is None:
+            # Simple version: remove the background value
+            data_corr = angleres.ARBackgroundSubtract(data)
+        else:
+            if data.shape != bg_image.shape:
+                # resize
+                bg_image = img.rescale_hq(bg_image, data.shape)
+            data_corr = img.Subtract(data, bg_image)  # metadata from data
+
+        return data_corr
+
+
+class ARRawProjection(ARProjection):
+    """
+    An ARProjection is a typical projection used to show 2D angle resolved images.
+    Its .image contains a DataArray 2D RGB (shape YXC), with metadata MD_PIXEL_SIZE and MD_POS.
+    The .background is still on the stream passed in the __init__ as the same background image
+    can be used for multiple projections connected to the same stream.
+    Depending on which type of stream is passed at creation, a more suitable subclass of
+    RGBSpatialProjection might be created (via the use of the __new__ operator).   # TODO still True??
+    That is the recommended way to create a RGBSpatialProjection.
+    """
+
+    def __init__(self, stream):
+        """
+        :param stream: (Stream) The stream the projection is connected to.
+        """
+        super(ARRawProjection, self).__init__(stream)
+
+        # Cached conversion of the detector image to polar representation
+        # TODO: automatically fill it in a background thread - still a TODO???
+        self._polar_dict = {}  # dict tuple (float, float, str or None) -> DataArray
+
+        if self.stream._pos.keys()[0][-1]:
+            self.polarization = self.stream.polarization  # make it an attribute of the projection
+            self.polarization.subscribe(self._onPolarization)
+
+    def _project2Polar(self, pos):
+        """
+        Return the polar projection of the image at the given position.
+        :param pos: (float, float, string or None) position (must be part of the .stream._pos)
+        :returns: (DataArray) The polar projection
+        """
+        # Note: Need a copy of the link to the dict. If self._polarimetry_dict is reset while
+        # still running this method, the dict might get new entries again, though it should be empty.
+        polar_dict = self._polar_dict
+        # TODO BUg happens sometimes
```

Comment:
That's probably because the RGBProjection immediately calls `_shouldUpdateImage()` after `__init__()`, and you only set `polar_dict` after the `super.__init__()`. 
One way would be to move `self._polar_dict = {} to before the `super.__init__()`.
However, looking at all the RGBProjection inheriting classes, it seems that they all call  `_shouldUpdateImage()` too. So probably the best is to remove the call to `_shouldUpdateImage()` from the `RGBProjection.__init__()`, and add it at the end of your `__init__()`.

Reply:
So do you mean to delete the line in the code of the init or somehow by a magic attribute method "del" or similar....If I delete the line I would have to add this line also in other projections.

Reply:
Yes, just delete physically the line calling `self._shouldUpdateImage()` in RGBProjection. Indeed, to compensate you'll need to add it to all the children classes, which don't have this call yet. I think it's only the ARProjections which don't have it yet. So that should be easy ;-)

Reply:
Done! merged it with update 2 and force pushed the new version